File: other_languages/javascript/linearSearch.js
const int_array = [1, 3, 5, 8, 12, 90, 33, 1, 5, 76, 32];
const args = process.argv.slice(2);

function linearSearch (sequence, target) {
    console.log("Linear Search");
    for (let i = 0; i < sequence.length; i++) {
        console.log(`Checking position ${i}`);
        if (sequence[i] === target) {
            console.log(`${target} found at position ${i}`);
            return;
        };
    };
    console.log(`${target} not found at any position`);
}

linearSearch(int_array, parseInt(args[0]));

File: other_languages/python/linear_search.py
import sys
from shared_sort import int_array


def main(args):
    try:
        to_find = int(args[0])
        linear_search(int_array, to_find)
    except IndexError:
        print("usage: python linear_search.py n")
        sys.exit(1)


def linear_search(sequence, target):
    print("Linear Search")
    for index, value in enumerate(sequence):
        print(f"Checking position {index}")
        if value == target:
            print(f"Found at position {index}")
            return
    print(f"{target} not found at any position")


if __name__ == '__main__':
    main(sys.argv[1:])
